src/TextField/TextField.js

```diff
@@ -177,6 +182,10 @@ class TextField extends Component {
      */
     inputStyle: PropTypes.object,
     /**
+     * Whether the input value is required.
+     */
+    isRequired: PropTypes.bool,
```

Comment:
Have you thought about using the `required` name?
It's closer to the HTML specification, shorter and already sementicaly express a *state*.